Objects/object.c

```diff
@@ -1715,10 +1715,16 @@ _PyObject_GenericGetAttrWithDict(PyObject *obj, PyObject *name,
             dict = (PyObject *)_PyObject_GetManagedDict(obj);
         }
         else {
+            Py_BEGIN_CRITICAL_SECTION(obj);
+#ifdef DISABLE_GIL
+            PyObject **dictptr = _Py_atomic_load_ptr(_PyObject_ComputedDictPointer(obj));
```

Comment:
This is the wrong thing to load atomically--dereferencing it should be the atomic operation. In fact, this doesn't do what you think it does: `_Py_atomic_load*` takes a pointer, and the thing that it points to gets atomically loaded and returned.